docs/mockwbemserver.rst

```diff
@@ -867,13 +794,236 @@ This adds the instances to the repository display of the previous example:
+Within the fake  WBEM server the response to client requests is normally
+determined by standard server responder methods that use data in the CIM
+repository to generate responses (ex. ``GetInstance`` gets the instances from the
+repository, possibly filters properties and attributes and returns the
+instance). However, it is expected that the user may want specific classes to
+generate responses that have side effects, or manage the returns differently
+than the normal responses.  Thus, for example, the DMTF defined CIM_Namespace
+class returns instances of each of the namespaces in the CIM repository. That means
+that somewhere an instance the class must be created each time a namespace is
+created or the responder must get the set of namespaces from the CIM repository
+to generate responses to ``GetInstance``, ``EnumerateInstance``, etc.
+
+Also, the :meth:`pywbem_mock.FakedWBEMConnection.InvokeMethod` client operation
+depends on a specific responder method in the WBEM server since the behavior of
+WBEM server methods is dependent on the specific method.
+
+To meet these needs, pywbem_mock has implemented the capability to create
+:term:`user-defined providers <user-defined provider>` that become request
+responders for defined classes and operations which can can manipulate the
+client input and generate responses specific for defined CIM classes.
+
+User-defined providers can be written by the user by defining subclasses of
+specific provider classes used by the fake WBEM server and registering these
+classes as providers using
+:meth:`~pywbem_mock.FakedWBEMConnection.register_provider`.
+
+Pywbem defines user-defined provider types so that specific request operations
+can be executed for each provider type.
+
+
+.. table:: Pywbem_mock provider types
+
+    ====================== ============ ========================  ================================
+    Provider Type          type name    CIM Request Operations    Default provider class
+    ====================== ============ ========================  ================================
+    method                 "method"      InvokeMethod             :ref:`Method Provider`
+    instance write         "instance"    CreateInstance           :ref:`Instance Write Provider`
+    instance write         "instance"    ModifyInstance           :ref:`Instance Write Provider`
+    instance write         "instance"    DeleteInstance           :ref:`Instance Write Provider`
+    ====================== ============ ========================  ================================
+
+Each user-defined provider is a Python class which is a subclass
+of the corresponding default provider for the provider type as defined in
+the previous table.
+
+
+.. _`Creating user-defined instance providers`:
+
+Creating user-defined instance providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ Instance user-defined providers can override the default implementations for
```

Comment:
Change "Instance user-defined providers" to "User-defined instance providers".

Reply:
DONE